OpenRA.Mods.Common/Traits/Render/ProductionBar.cs

```diff
@@ -67,7 +67,7 @@ void INotifyCreated.Created(Actor self)
 
 		void ITick.Tick(Actor self)
 		{
-			var current = queue.CurrentItem();
+			var current = queue.AllQueued().OrderBy(i => i.RemainingTime).FirstOrDefault();
```

Comment:
This needs to ignore units that haven't started yet!

It breaks the production bar if you are are producing a long build time unit (e.g. a mammoth tank in RA) and then queue a unit that has a total build time shorter than what is left on the current item (e.g. a flak truck).